Homework-№4-from-28.09.2022/task 1.c

```diff
@@ -0,0 +1,183 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#define size 32 //in this task all lengths are a 32
+
+void printOfArray(char *array)
+{
+    for (int i = 0; i < size; ++i)
+    {
+        printf("%c", array[i]);
+    }
+    printf("\n");
+}
+
+void initialization(char *array)
+{
+    for (int i = 0; i < size; ++i)
+    {
+        array[i] = '0';
+    }
+}
+
+void binAddingOne(char *binOfNum)
+{
+    bool notExit = true;
+    int i = size - 1;
+    while (notExit)
+    {
+        if (binOfNum[i] == '1')
+        {
+            --i;
+        }
+        else
+        {
+            notExit = false;
+            for (int j = i + 1; j < size; j++)
+            {
+                binOfNum[j] = '0';
+            }
+            binOfNum[i] = '1';
+        }
+    }
+}
+
+void reverse(char *array)
+{
+    for (int i = 0; i < size; ++i)
+    {
+        switch (array[i])
+        {
+            case '0':
+                array[i] = '1';
+                break;
+            default:
+                array[i] = '0';
+                break;
+        }
+    }
+}
+
+void toBin(int number, char *binRepr)
+{
+    int temp = number;
+    temp = abs(temp);
+    int pointer = size - 1;
+    while (temp > 0)
+    {
+        if (temp % 2 == 0)
+        {
+            binRepr[pointer] = '0';
+        }
+        else
+        {
+            binRepr[pointer] = '1';
+        }
+        temp /= 2;
+        --pointer;
+    }
+    if (number < 0)
+    {
+        reverse(binRepr);
+        binAddingOne(binRepr);
+    }
+}
+
+int fromBin(char *binRepr)
+{
+    int number = 0;
+    int temp = 1;
+    bool isItOne = true;
```

Comment:
```suggestion
    bool isNegativeNumber = true;
```
Вроде, такой смысл у этой переменной